content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,354 @@
+---
+Title: "Bulding a ToDo List with Ruby on Rails"
+Description: "Learn how to build a ToDo List with Ruby on Rails"
+DatePublished: "2022-04-28"
+Categories:
+  - "computer-science"
+  - "web-development"
+Tags:
+  - "Ruby on Rails"
+  - "Ruby"
+  - "PostgreSQL"
+  - "Database"
+CatalogContent:
+  - "learn-ruby"
+  - "learn-rails"
+---
+
+_**Prerequisites:** Ruby, Ruby on Rails, PostgreSQL_
+
+_**Versions:** Ruby 3.2.0, Ruby on Rails 7.0.4, PostgreSQL 15_
+
+## Introduction
+
+This article is an introduction to building a full-stack app using Ruby on Rails. For this article, we are building a To-do list.
+
+I was inspired to write this article while learning Ruby on Rails. I often got lost looking for information on the official Rails guide and I want to help you reduce the time it’ll take you to get started with building your own Rails app.
+
+The article is beginner friendly, and I hope it helps you learn something new.
+
+## What I’ll talk about:
+
+1. Setting up a new Rails app.
+2. Understanding the MVC design pattern.
+3. Creating a new schema in your local database using migrations.
+4. Handling requests within controller methods.
+5. Using ERB to write logic and display content in HTML.
+6. Understanding the Rails routing system.
+
+## Brief overview of Ruby on Rails
+
+Rails is a web application development framework written in the Ruby programming language. It is an easy framework to get started with because it allows you to write less code while accomplishing more than many other languages and frameworks.
+
+The Rails framework is guided by two major principles:
+
+1. Don’t Repeat Yourself: The [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) is very common in software development. The idea is that developers should spend less time writing code by avoiding repetition. This is done by making sure every element of your code can stand alone and be called anywhere, and changes to that element don't have to be repeated across several points.
+2. Convention over Configuration: This is one of Rails’ superpowers. Rails has a set of guidelines and format for doing things. From naming methods, classes, files, and building a database schema. These conventions help to keep the whole application in sync and help you spend less time debugging errors that result from the wrong configuration.
+
+## Setting up your dev environment: Installing Ruby, Rails, and PostgreSQL
+
+[GoRails](https://goRails.com/) Has in-depth guides on setting up Rails on various machines. Here's a link to all of them. Since we're using PostgreSQL for this project, skip any step that mentions setting up SQLite or MySQL.
+
+- [For Ubuntu](https://goRails.com/setup/ubuntu/22.04)
+- [Mac OS](https://goRails.com/setup/macos/13-ventura)
+- [For Windows](https://www.hanselman.com/blog/ruby-on-Rails-on-windows-is-not-just-possible-its-fabulous-using-wsl2-and-vs-code)
```

Comment:
Not sure if we should be linking to outside sources when we have: https://www.codecademy.com/resources/docs/ruby/ruby-on-rails

And also the official documentation sites which cover the installation process:
- https://guides.rubyonrails.org/getting_started.html#creating-a-new-rails-project-installing-rails
-

Reply:
@deyemiobaa We may also need to mention postgresql installation since we're using it below

Reply:
@yangc95 Those external resources go into more detail on how to install Rails with PostgreSQL. They are well-known resources and are regularly updated to support several operating systems.

I didn't mention PostgreSQL installation because the resources I linked already discussed them.